src/main/perl/Lock.pm

```diff
@@ -73,172 +65,80 @@ The B<CAF::Lock> class provides methods for handling application locking.
-    return undef;
-  }
-  unless ($lf->close()) {
-    $self->error("cannot close lock file: ".$self->{'LOCK_FILE'});
-    return undef;
-  }
-  if ($pid !~ m{^(\d+)$}) {
-      $self->error("Strange PID $pid holding lock $self->{LOCK_FILE}");
-      return undef;
-  }
-  return $1;
-}
-
-
-=pod
-
-=item is_stale()
-
-Returns SUCCESS if the lock is stale - a lock file is set but the
-corresponding PID does not exist. Returns undef otherwise.
-
-=cut
-
-sub is_stale {
-  my $self=shift;
-
-  return undef unless ($self->is_locked());
-  my $lock_pid=$self->get_lock_pid();
-  if ($lock_pid && kill(0, $lock_pid)) {
-      return undef;
-  } else {
-      return SUCCESS;
-  }
-}
-
-
-=pod
+=item set_lock(I<retries>, I<timeout>, I<force>)
 
-=item set_lock ($retries,$timeout,$force);
-
-Tries $retries times to set the lock. If $force is set to FORCE_NONE
+Tries I<retries> times to set the lock.  If I<force> is set to B<FORCE_NONE>
 or not defined and the lock is set, it sleeps for
-rand($timeout). Writes the current PID ($$) into the lock
-file. Returns SUCCESS or undef on failure.
+rand(I<timeout>).  Returns B<SUCCESS>, or B<undef> on failure.
 
-If $retries or $timeout are not defined or set to 0, only a single
+If I<retries> or I<timeout> are not defined or set to 0, only a single
 attempt is done to acquire the lock.
 
-If $force is set to FORCE_ALWAYS then the lockfile is just set
-again, independently if the lock is already set by another application
-instance.
-
-If $force is set to FORCE_IF_STALE then the lockfile is set if the
-application instance holding the lock is dead (PID not alive).
-
-If $force is set to FORCE_ALWAYS, or if $force is defined to
-FORCE_IF_STALE and a stale lock file is detected, then neither
-$timeout nor $retries are taken into account.
+If I<force> is set to B<FORCE_ALWAYS> then the lock file is just set
+again, even if the lock is already set by another application
+instance, and neither I<timeout> nor I<retries> are taken
+into account.
 
 =cut
 
 sub set_lock {
-  my ($self,$retries,$timeout,$force)=@_;
+  my ($self, $retries, $timeout, $force) = @_;
 
-  $retries=0 unless (defined $retries);
-  $timeout=0 unless (defined $retries);
+  $retries = 0 unless (defined $retries);
+  $timeout = 0 unless (defined $retries);
 
   if ($self->{LOCK_SET}) {
     # oops.
     $self->error("lock already set by this application instance");
-    return undef;
+    return;
   }
 
-  unless ($force == FORCE_ALWAYS ||
-	  ($force == FORCE_IF_STALE && $self->is_stale())) {
-    my $tries=1;
-    while ($tries <= $retries &&
-	   ($self->is_locked() && !($force == FORCE_IF_STALE && $self->is_stale()))) {
-      my $sleep=rand($timeout);
-      $self->verbose("lockfile is already held, try $tries out of $retries");
-      $tries++;
-      sleep($sleep);
+  my $tries = 0;
+  do {
+    if ($tries > 0) {
+      $self->verbose("lock file is already held, try $tries out of $retries");
+      sleep(rand($timeout));
```

Comment:
for one: it was always like that.
two: it could make sense to do this in a distributed environment.

Also if there are multiple processes waiting for some resource that is protected by the lock randomization makes sense (to me).